cmake/IRODSConfigVersion.cmake.in

```diff
@@ -0,0 +1,170 @@
+#   as potentially compatible if the find_package()-calling project is searching for the next
+#   minor release or any earlier version of the same major release.
+
+set(IRODS_VERSION_1 "@IRODS_VERSION_MAJOR@")
+set(IRODS_VERSION_2 "@IRODS_VERSION_MINOR@")
+set(IRODS_VERSION_3 "@IRODS_VERSION_PATCH@")
+set(PACKAGE_VERSION "${IRODS_VERSION_1}.${IRODS_VERSION_2}.${IRODS_VERSION_3}")
+
+# Get rid of any leading zeroes. (There probably won't ever be any, but best to check anyway.)
+if(NOT IRODS_VERSION_1 VERSION_EQUAL "0")
+  string(REGEX REPLACE "^0+" "" IRODS_VERSION_1 "${IRODS_VERSION_1}")
+endif()
+if(NOT IRODS_VERSION_2 VERSION_EQUAL "0")
+  string(REGEX REPLACE "^0+" "" IRODS_VERSION_2 "${IRODS_VERSION_2}")
+endif()
+if(NOT IRODS_VERSION_3 VERSION_EQUAL "0")
+  string(REGEX REPLACE "^0+" "" IRODS_VERSION_3 "${IRODS_VERSION_3}")
+endif()
+
+# Check version range first.
+if (PACKAGE_FIND_VERSION_RANGE)
+  # Here we check if the iRODS version is outside the version range passed to find_package().
+  # This is a bit of a hefty conditional, so let's break it down.
+  # PACKAGE_FIND_VERSION_MIN is the lower end of the version range, and
+  # PACKAGE_FIND_VERSION_RANGE_MIN tells us whether the lower end of the version range is
+  # inclusive ("INCLUDE") or exclusive ("EXCLUDE"). (At time of writing, CMake only supports
+  # "INCLUDE" for the lower end, but the way the documentation is written implies that "EXCLUDE"
+  # might someday be supported, and supporting that possibility here is trivial, so we do.)
+  # PACKAGE_FIND_VERSION_MAX is the upper end of the version range, and
+  # PACKAGE_FIND_VERSION_RANGE_MAX tells us whether the upper end of the version range is
+  # inclusive ("INCLUDE") or exclusive ("EXCLUDE").
+  if (
+    # lower end of version range is inclusive and iRODS version is older than requested range
+    ((PACKAGE_FIND_VERSION_RANGE_MIN STREQUAL "INCLUDE" AND PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION_MIN)
+      # lower end of version range is exclusive and iRODS version is older than or equal to requested range
+      OR (PACKAGE_FIND_VERSION_RANGE_MIN STREQUAL "EXCLUDE" AND PACKAGE_VERSION VERSION_LESS_EQUAL PACKAGE_FIND_VERSION_MIN))
+    # upper end of version range is inclusive and iRODS version is newer than requested range
+    OR ((PACKAGE_FIND_VERSION_RANGE_MAX STREQUAL "INCLUDE" AND PACKAGE_VERSION VERSION_GREATER PACKAGE_FIND_VERSION_MAX)
+      # upper end of version range is exclusive and iRODS version is newer than or equal to requested range
+      OR (PACKAGE_FIND_VERSION_RANGE_MAX STREQUAL "EXCLUDE" AND PACKAGE_VERSION VERSION_GREATER_EQUAL PACKAGE_FIND_VERSION_MAX))
+  )
+    set(PACKAGE_VERSION_COMPATIBLE FALSE)
+  else()
+    set(PACKAGE_VERSION_COMPATIBLE TRUE)
+  endif()
+elseif (PACKAGE_FIND_VERSION VERSION_EQUAL PACKAGE_VERSION) # the easy case: an exact match
+  set(PACKAGE_VERSION_EXACT TRUE)
+  set(PACKAGE_VERSION_COMPATIBLE TRUE)
+else()
```

Comment:
The comments on the other two cases were very helpful (at least for me). Can we include one for the else case? It's not clear to me what the else means other than "not exact match" and "not in a range".

Maybe it's hard to define because there's a variety of things that it could mean? If that's the case, don't worry about it. Just thought it would be good to include.

Reply:
Yeah, the `else` is for literally any other case, so it's a wide variety of potential cases.